pkg/operator/controller/nodeobservability/controller.go

```diff
@@ -52,41 +56,120 @@ const (
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,namespace=node-observability-operator,resources=roles,verbs=list;get;create;watch;delete;
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,namespace=node-observability-operator,resources=rolebindings,verbs=list;get;create;watch;delete;
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+
+// New creates the nodeObservability controller from mgr and cfg. The controller will be pre-configured
+// to watch for nodeObservability objects across all namespaces.
+func New(mgr manager.Manager, cfg Config) (controller.Controller, error) {
+	log := ctrl.Log.WithName(controlleroperator.ControllerName)
+
+	r := &NodeObservabilityReconciler{
+		config: cfg,
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+		Log:    zap.New(zap.UseDevMode(true)),
+	}
+
+	c, err := controller.New(controlleroperator.ControllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return nil, err
+	}
+
+	if err := c.Watch(&source.Kind{Type: &operatorv1alpha2.NodeObservability{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return nil, err
+	}
+	if err := c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &operatorv1alpha2.NodeObservability{},
+	}); err != nil {
+		return nil, err
+	}
+
+	// enqueue all nodeObservability instances if the kubelet CA config map changed
+	// EnqueueRequestForOwner won't work here
+	// because the kubelet CA configmap doesn't belong to any particular nodeObservability instance
+	allNodeObsInstances := func(o client.Object) []reconcile.Request {
+		nodeObservabilityList := &operatorv1alpha2.NodeObservabilityList{}
+		requests := []reconcile.Request{}
+		if err := mgr.GetCache().List(context.Background(), nodeObservabilityList); err != nil {
+			log.Error(err, "failed to list nodeObservability for kubelet CA configmap")
+			return requests
+		}
+		for _, ed := range nodeObservabilityList.Items {
+			log.Info("queueing nodeObservability for kubelet CA configmap", "name", ed.Name)
+			request := reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name: ed.Name,
+				},
+			}
+			requests = append(requests, request)
+		}
+		return requests
+	}
+	if err := c.Watch(
+		&source.Kind{Type: &corev1.ConfigMap{}},
+		handler.EnqueueRequestsFromMapFunc(allNodeObsInstances),
+		// only the target kubelet CA configmap
+		predicate.NewPredicateFuncs(ctrlutils.InNamespace(r.Namespace)),
+		predicate.NewPredicateFuncs(ctrlutils.HasName(controlleroperator.NodeObsKubeletCAConfigMapName(r.Namespace).Name)),
+	); err != nil {
+		return nil, err
+	}
```

Comment:
This is goo but it's part of the next PR for which you created a subtask: hash annotation in the daemonset.